docs/src/architecture/09_architecture_decisions/0007-api-versions.md

```diff
@@ -0,0 +1,97 @@
+
+1. All endpoints URIs will be under the `/api` path in the backend.
+2. The general structure of any endpoint will be `/api/<version>/<path>`
+3. `<version>` defines a version of the endpoint supplied by a `<path>` and is **NOT** aligned with any other versions.
+4. `<version>` **MUST** either be `draft` or matching the regular expression `^v(?!0)\d{1,}$`[^1]
+5. `<path>` Can be any path of at least one element as required for the endpoint.
+6. **ALL** new endpoints start as `draft` they are promoted to a versioned endpoint by a deliberate decision.
+7. New endpoints **MUST NOT** be versioned other than `draft` in their initial PR.
+8. New endpoints can only be versioned after the behavior and form of the endpoint has been validated, in a subsequent PR.
+9. *ALL* draft endpoints are subject to change and can be safely modified in any way.
+10. `draft` Versioned endpoints can be used by frontend code.  
+However, if the API changes or disappears, the front end should fail gracefully.
+11. The numbered version of an endpoint will always start at `v1` and increments sequentially.
+12. API versions are **ONLY** incremented if their behavior changes in a way that can reasonably be considered a breaking change
+with respect to the currently published OpenAPI specification for that Endpoint.
+13. The API endpoint versions are not aligned to any semantic versioning of the backend.
+14. `draft` endpoints should not be tested in CI, in such a way as they break the CI pipeline if they change.
+15. **ALL** versioned endpoints **SHOULD** have CI tests which ensure the API endpoint itself has not changed in a breaking way.
+16. Code generation can generate code for `draft` endpoints, provided doing so does not result in breaking CI.
+17. Any Integration tests written against `draft` endpoints should not fail.  
+They may produce warnings if they do not match expected outputs.
+18. Re-versioning an endpoint is NOT required if the change to it is backward compatible with the existing endpoint.
+A non-exhaustive list of possible cases for this are:
+    * A new *OPTIONAL* query parameter is added to an endpoint.
+    * A response field is added, and the OpenAPI document defined that `"additionalProperties": true`
+in the schema definition where the additional field is to be added.
+    * A previously undocumented behavior is documented.  
+In this case, the behavior must already exist, it is not a breaking change to clarify documentation.
+19. *ALL* non-breaking changes proposed in a PR to existing API's,
+or migrating an API from `draft` to a versioned API **MUST** be signed off by
+    * at least 1 Architect on the Team; and
+    * the Engineering manager; and preferably
+    * A senior member of the team with primary responsibility for the frontend.
+20. When a new version of an endpoint is released, the existing endpoint **MUST** be marked as `deprecated`
+in the OpenAPI specification for the endpoint.
```

Comment:
I find it difficult to understand how this should work. Could you please add an example?

Reply:
I added a link to how OpenAPI manages deprecation.
When we formalize deprecation, we would need another ADR which explains it.
There are a few options we can consider in order to document it, and advise consumers of our API about whats happened.